themes/default/content/what-is/what-is-google-secret-manager.md

```diff
@@ -0,0 +1,113 @@
+
+Enter authorization code: 4/0A....
+
+You are now logged in as [your@email.com].
+Your current project is [your-project].  You can change this setting by running:
+  $ gcloud config set project PROJECT_ID
+```
+
+### Create a secret via the CLI
+
+You can create a secret by running the `gcloud secrets create` command:
+
+```bash
+$ gcloud secrets create my-secret
+
+Created secret [my-secret].
+```
+
+Now add your secret data to the secret as shown below:
+
+```bash
+$ echo -n "my-secret-data" | gcloud secrets versions add my-secret --data-file=-
+
+Created version [1] of the secret [my-secret].
+```
+
+Verify the secret was created with the `gcloud secrets list` command.
+
+```bash
+$ gcloud secrets list
+
+NAME         CREATED              REPLICATION_POLICY  LOCATIONS
+api-key      2023-10-09T09:04:30  automatic           -
+my-secret    2023-11-30T14:05:58  automatic           -
+test-secret  2023-11-10T09:40:29  automatic           -
+```
+
+{{< notes type="info" >}}
+
+Many infrastructure as code platforms, including Pulumi, have support for creating secrets. You can learn more about how to create and manage secrets in Pulumi by taking a look at [Pulumi Secrets documentation](/docs/concepts/secrets/).
+
+{{< /notes >}}
+
+## Accessing Azure Key Vault secrets
+
+Now that you have created a Google Secret Manager secret, you can access the value via the gcloud CLI using the `gcloud secrets versions access` command.
+
+```bash
+$ gcloud secrets versions access latest --secret="my-secret"
+
+my-secret-data
+```
+
+## Challenges and considerations
+
+Google Secret Manager, while a powerful tool for managing secrets and cryptographic keys, does come with its own set of challenges, considerations, and limitations. Some of the key aspects to be aware of include:
+
+- **Compatibility and integration**: Google Secret Manager is optimized for workloads in Google Cloud. For organizations operating in a multi-cloud or hybrid cloud environment, integrating Secret Manager with other cloud platforms might not be straightforward. In such cases, using a third-party tool like [Pulumi ESC](/docs/pulumi-cloud/esc/) might be more effective.
+- **Quotas and limits on secret access**: When retrieving secrets frequently, such as in SaaS applications that make API calls, it's important to ensure that project-level quotas are sufficient. Secret Manager has a limit of 90,000 access requests per minute per project, which can be increased via a Google Cloud support ticket. Additionally, there are limits on the size of the secret value (64 KiB) and the number of aliases a secret can have (50 aliases).
+- **Lack of auto-rotation for secrets**: Currently, Google Secret Manager does not have the capability to auto-rotate secrets. This means organizations need to implement their own mechanisms or use infrastructure-as-code (IaC) tools such as [Pulumi](/docs/get-started/) to ensure secrets are rotated in compliance with their security policies.
```

Comment:
Link to the iac page

I dont get why iac and Pulumi can ensure secrets are rotated. It just means they dont support oidc right? In that case its not really connected to iac as the solution? You should phrase they need to implement dynamic secrets rotation or use oidc. You can use ESC to do this, etc, etc